wcfsetup/install/files/js/WoltLabSuite/Core/Ui/File/Upload.js

```diff
@@ -0,0 +1,512 @@
+					
+					elBySel('small', this._fileElements[uploadId][i]).innerHTML = '';
+					elBySel('.icon', this._fileElements[uploadId][i]).classList.remove('fa-spinner');
+					elBySel('.icon', this._fileElements[uploadId][i]).classList.add('fa-ban');
+					
+					if (elBySel('.innerError', this._fileElements[uploadId][i]) === null) {
+						var innerError = elCreate('span');
+						innerError.classList = 'innerError';
+						innerError.textContent = data['error'][i].errorMessage;
+						DomUtil.insertAfter(innerError, elBySel('small', this._fileElements[uploadId][i]));
+					}
+					else {
+						elBySel('.innerError', this._fileElements[uploadId][i]).textContent = data['error'][i].errorMessage;
+					}
+				}
+				else {
+					throw new Error('Unknown uploaded file for uploadId '+ uploadId +'.');
+				}
+			}
+			
+			// create delete buttons
+			this._deleteHandler.rebuild();
+			this.checkMaxFiles();
+		},
+		
+		/**
+		 * File input change callback to upload files.
+		 *
+		 * @param	{Event}		event		input change event object
+		 * @param	{File}		file		uploaded file
+		 * @param	{Blob}		blob		file blob
+		 * @return	{(int|Array.<int>|null)}	identifier(s) for the uploaded files
+		 */
+		_upload: function(event, file, blob) {
+			// remove failed upload elements first
+			var failedUploads = DomTraverse.childrenByClass(this._target, 'uploadFailed');
+			for (var i = 0, length = failedUploads.length; i < length; i++) {
+				elRemove(failedUploads[i]);
+			}
+			
+			var uploadId = null;
+			
+			var files = [];
+			if (file) {
+				files.push(file);
+			}
+			else if (blob) {
+				var fileExtension = '';
+				switch (blob.type) {
+					case 'image/jpeg':
+						fileExtension = '.jpg';
+						break;
+					
+					case 'image/gif':
+						fileExtension = '.gif';
+						break;
+					
+					case 'image/png':
+						fileExtension = '.png';
+						break;
+				}
+				
+				files.push({
+					name: 'pasted-from-clipboard' + fileExtension
+				});
+			}
+			else {
+				files = this._fileUpload.files;
+			}
+			
+			if (files.length && files.length + this.countFiles() <= this._options.maxFiles) {
+				if (this._options.singleFileRequests) {
+					uploadId = [];
+					for (var i = 0, length = files.length; i < length; i++) {
+						var localUploadId = this._uploadFiles([ files[i] ], blob);
+						
+						if (files.length !== 1) {
+							this._multiFileUploadIds.push(localUploadId)
+						}
+						uploadId.push(localUploadId);
+					}
+				}
+				else {
+					uploadId = this._uploadFiles(files, blob);
+				}
+			}
+			else {
+				var innerError = elBySel('small.innerError:not(.innerFileError)', this._buttonContainer.parentNode);
+				
+				if (innerError === null) {
+					innerError = elCreate('small');
+					innerError.classList = 'innerError';
+					DomUtil.insertAfter(innerError, this._buttonContainer);
+				}
+				
+				innerError.textContent = WCF.Language.get('wcf.upload.error.reachedRemainingLimit').replace(/#remaining#/, this._options.maxFiles);
```

Comment:
Use `Language.get()` instead and may-be, as this a new API, add a new language item does not use `#…#` but proper variables.